src/notes/FluxNotesEditor.jsx

```diff
@@ -420,15 +420,32 @@ class FluxNotesEditor extends React.Component {
         let transform;
         transform = this.state.state.transform();
 
-        shortcut.clearValueSelectionOptions();
-        shortcut.setText(selection.context);
-        if (shortcut.isContext()) {
-            shortcut.setValueObject(selection.object);
-            this.contextManager.addShortcutToContext(shortcut);
-            this.contextManager.contextUpdated();
+
+        if (shortcut.completionComponent.name === "ContextGetHelp") {
```

Comment:
I don't think we want FluxNotesEditor to have to worry about different shortcuts and their completionComponents. Instead, we might make this logic a part of the getHelp completion component. In the place where Sarah defined an Expand function previously. This logic should be used there to determine how to insertText based on the current shortcut. The selection of the expand dropDown element should trigger the execution of that Expand function, and then feeding the result of that Expand function into a function that replaces the current shortcut with the new expanded text. What are people's thoughts on that approach?

Reply:
I agree with this ^

Reply:
Updated